core/models/application.py

```diff
@@ -50,9 +50,13 @@ def get_users_from_access_list(self):
         - Returns a list of Users who passed the test
         """
         from core.models import AtmosphereUser
-        all_users = AtmosphereUser.objects.all()
+
+        all_users = AtmosphereUser.objects.none()
         for pattern_match in self.access_list.all():
-            all_users &= pattern_match.validate_users()
+            if pattern_match.allow_access:
+                all_users |= pattern_match.validate_users()
+            else:
+                all_users &= pattern_match.validate_users()
```

Comment:
the &= case won't work, logic should be like: if allow_access all_users begins as the empty set, else all_users begins as everyone

Reply:
Submitted test cases, the logic works as expected for now. Future changes can come in a separate PR